### ISSUE TITLE
How to start and evaluate trials on different pods

### ISSUE THREAD

**Issue by bernardohenz** [drosin/argo-hyperparameter-tuner#1]
Hi,
I'm learning to work with argo+optuna, and yous repository is a **gem**.

Right now I have multiple models being trained (in parallel, using argo workflows) that are later combined and evaluated on a testset.
I'm trying to use optuna for hyperparameter tuning.
Although I could optuna to optimize each training by itself, I would like that a single trial consists the training of all models.
For this case, I would have to create a trial before calling the training scripts (and use `trial.suggest...` inside of them), and  later evaluate the loss on the evaluate script.

Could you help me on how can I 'breakdown' the `study.optimize(` for creating the trial on one script (before training scripts), and evaluating it in another script (on the one that performs the emsemble)?

**Comment by drosin** [comment]
Hi @bernardohenz , thanks so much for your feedback :-) 
So what is missing when you run this argo workflow? hyperparameter-tuner-optuna-level.yaml

**Comment by bernardohenz** [comment]
I would like to do something in the sense:
```
  templates:
    - name: train-model-A
       script:
          ....
        command: [python]
        source: |
          .....
          
          model.fit(x_train, y_train, batch_size=trial.suggest_int('batchsizeA', 8, 64),  epochs=trial.suggest_int('epochsA', 8, 64))
          model.save('model-a')
    - name: train-model-B
       script:
          ....
        command: [python]
        source: |
          .....
          
          model.fit(x_train, y_train, batch_size=trial.suggest_int('batchsizeB', 8, 64),  epochs=trial.suggest_int('epochsB', 8, 64))
          model.save('model-b')
    - name: evaluate
       script:
        ...
        command: [python]
        source: |
          .....
          
          model_a = models.load_model('model-a')
          model_b = models.load_model('model-b')
          def local_objective(trial):
                y_pred_a = model_a.predict(x_test)
                y_pred_b = model_b.predict(x_test)
                lambda = trial.suggest_float("emsemble_weight", 0.1, 0.9)
                y_pred = lambda * y_pred_a  + (1-lambda) * y_pred_b
                return binary_crossentropy(y_test, y_pred)
          study.optimize(local_objective, n_trials=100)

```

I don't know to make a trial to run each training script, followed by the evaluation script.
Also, how can I reference the same trial on each one of the scripts.
I've seen that a `optuna.study` created in one script, can be loaded into another, can the same be done with trials?

**Comment by drosin** [comment]
I understand now. This will require some additional work. Maybe you could achieve it with the Ask-and-Tell Interface of Optuna? https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/009_ask_and_tell.html

**Comment by bernardohenz** [comment]
Hey @drosin , thanks for the tip.

Using `ask` and `tell` I can `trial.suggest` on training, and `trial.tell` on evaluate.
I still do not know how to recover an existing trial (`trial.ask()` creates a new trial, but I want `train-model-A` and `train-model-B` using the *same* trial). I've tried `study.get_trials()` or `study.trials`, but it returns `FrozenTrial`, which cannot be used for `trial.suggest_....`


Do you have any idea how to handle this?

**Comment by drosin** [comment]
Would it help to 
1) create another step in the workflow that creates the trial 
2) make this trial available to other steps as output parameter
3) have multiple other steps use these trial parameters

Hope, this helps!
